### ISSUE TITLE
Cannot build on an encrypted file system (ecryptfs)

### ISSUE THREAD

**Issue by vedran6** [ACINQ/eclair#379]
Using Ubuntu 14.04 with an ecrypted home directory and getting this error when trying to build:

```
[ERROR] /home/vedran/eclair/eclair-core/src/main/scala/fr/acinq/eclair/api/Service.scala:180: error: File name too long
[ERROR] This can happen on some encrypted or legacy file systems.  Please see SI-3623 for more details.
[ERROR]                       case "allchannels"  => completeRpcFuture(req.id, (router ? 'channels).mapTo[Iterable[ChannelAnnouncement]].map(_.map(c => ChannelInfo(c.shortChannelId.toHexString, c.nodeId1, c.nodeId2))))
[ERROR]                                                      
```

Is there a straightforward workaround?

**Comment by verretor** [comment]
Filenames are too long for ecryptfs (max about 140 chars). The only way that I know works is compiling on a file system that isn't encrypted.

It's actually how I compile it and I don't like it.

**Comment by sstone** [comment]
Yes, this a limitation of the scala 2.10 compiler and there are no generic solutions (but you can experiment with the `-Xmax-classfile-name` compiler flag, I haven't tried yet). For now the simplest workaround, as explained above, is to compile from a directory that is not encrypted.

This affects people who have chosen to encrypt their home folders, but not people who use full disk encryption (which may arguably be a better option for single user systems such as laptops).
